src/routes/store/index.svelte

```diff
@@ -1,18 +1,89 @@
 <script context="module">
+  import * as api from '@/utils/api.js';
+  const productLimit = 24;
+
   export async function preload() {
-    const res = await this.fetch('/store.json');
-    return {items: await res.json()};
+    const { pages, data } = await api.get(`/products?limit=${productLimit}`);
+    const colors = await api.get('/colors');
+    return { products: data, colors, pages };
   }
 </script>
 
 <script>
+  import { stores } from '@sapper/app';
   import Layout from '@/layouts/default.svelte';
   import Tagline from '@/containers/store/tagline.svelte';
-  import StoreCard from '@/components/store/card.svelte';
+  import Balance from '@/components/store/balance.svelte';
   import Filters from '@/containers/store/filters.svelte';
-  import Ordering from '@/containers/store/ordering.svelte';
+  import StoreCard from '@/components/store/card.svelte';
+  import Pagination from '@/components/common/pagination.svelte';
+
+  const orderOptions = [
+    { orderBy: 'addition_time', order: 'asc' },
+    { orderBy: 'addition_time', order: 'desc' },
+    { orderBy: 'price', order: 'asc' },
+    { orderBy: 'price', order: 'desc' },
+  ];
+  const orderLabels = [
+    'newest first',
+    'newest last',
+    'cheapest first',
+    'cheapest last',
+  ];
+
+  const { session } = stores();
+
+  export let products;
+  export let pages;
+  export let colors;
+
+  let balance;
+  let order = orderOptions[0];
+  let orderLabel = orderLabels[0];
+  let currentPage = 1;
+
+  let filterElement;
+
+  try {
+    balance = $session.user.balance;
+  } catch (e) {
+    balance = 1;
+  }
 
-  export let items;
+  function updateProducts(filtering) {
+    let query = [];
+    if (filtering.searchQuery) {
+      query.push(`q=${filtering.searchQuery}`);
+    }
+    if (filtering.minPrice != null) {
+      query.push(`min=${filtering.minPrice}`);
+    }
+    if (filtering.maxPrice != null) {
+      query.push(`max=${filtering.maxPrice}`);
+    }
+    let excludedColors = filtering.excludedColors.map(color => color.slice(1));
+    if (filtering.colorlessExcluded) {
+      excludedColors.push('null');
+    }
```

Comment:
There is the utils/generateQuery module precisely to format the parameters in a query. no need to do it here manually